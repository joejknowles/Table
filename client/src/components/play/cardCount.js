import React from 'react';
import { connect } from 'react-redux';

import res from '../../resources/pages/play';

import { cardCountSelector } from '../../reducers';

export const CardCount = ({ count }) => (
  <div className="card-count">
    { res.cardCount(count) }
  </div>
);

const mapStateToProps = (state) => ({
  count: cardCountSelector(state)
});

export default connect(mapStateToProps)(CardCount);
