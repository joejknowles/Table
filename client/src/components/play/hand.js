import React from 'react';
import { connect } from 'react-redux';

import PlayablePile from './playablePile';
import StaticPile from './staticPile';
import NoCards from './noCards';

import { cardCountSelector } from '../../reducers';
import { myTurnSelector } from '../../selectors';

export const Hand = ({ cardCount, myTurn }) => {
  if (cardCount > 0) {
    return myTurn ? <PlayablePile /> : <StaticPile />;
  }
  return <NoCards />
};

const mapStateToProps = (state) => ({
  cardCount: cardCountSelector(state),
  myTurn: myTurnSelector(state)
});

export default connect(mapStateToProps)(Hand);
