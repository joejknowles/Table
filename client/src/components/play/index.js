import React from 'react';
import { connect } from 'react-redux';

import Hand from './hand';
import { handSelector } from '../../reducers';
import '../../styles/Play.css';
import res from '../../resources/pages/play';

export const Play = ({ hand }) => (
  <div className="App Play">
    <div className="deck">
      { hand > 0 ?
          <Hand /> :
          <p className='no-cards-message'>
            { res.noCardsMessage }
          </p>
      }
    </div>
  </div>
);


export default connect(
  (state) => ({ hand: handSelector(state) })
)(Play);
