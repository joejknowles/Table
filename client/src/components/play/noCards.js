import React from 'react';
import res from '../../resources/pages/play';

export default () => (
  <div className='pile'>
    <div className="noCards">
        <p className='no-cards-message'>
          { res.noCardsMessage }
        </p>
    </div>
  </div>
);
