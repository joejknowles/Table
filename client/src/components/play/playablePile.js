import React from 'react';
import PlayCardButton from './playCardButton';
import SwipeablePile from './swipeablePile';

function is_touch_device() {
  if ( 'callPhantom' in window ) return false; // integration tests fail without this
  return 'ontouchstart' in window ||        // works on most browsers
       navigator.maxTouchPoints;       // works on IE10/11 and Surface
};

export default () => (
    is_touch_device() ?
      <SwipeablePile />
        :
      <PlayCardButton />
);
