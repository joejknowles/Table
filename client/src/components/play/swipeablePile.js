import React from 'react';
import { connect } from 'react-redux';
import Cards, { Card } from '../../lib/swipe-cards';
import { cardCountSelector } from '../../reducers';

const swipeableCard = (playCard) => (
  <Card
    onSwipeTop={ playCard }
    key='top'>
    <div className='face-down'/>
  </Card>
);

export const SwipeablePile = ({ playCard, cardCount }) => (
  <Cards className="pile">
    {
      cardCount > 1 ?
      [
      swipeableCard(playCard)
    ,
    <Card key='bottom' children={ <div className='face-down'/> } />
    ] : [
        swipeableCard(playCard)
      ]
    }
  </Cards>
);

const mapStateToProps = (state) => ({
  cardCount: cardCountSelector(state)
});

const mapDispatchToProps = (dispatch) => ({ playCard: () => dispatch({ type: 'PLAY_CARD' }) })

export default connect(mapStateToProps, mapDispatchToProps)(SwipeablePile);
