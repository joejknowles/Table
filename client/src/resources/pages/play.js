export default {
  playCard: 'Play Card',
  noCardsMessage: 'No more cards'
};
