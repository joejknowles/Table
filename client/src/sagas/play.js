import { call, takeEvery } from 'redux-saga/effects';

import * as toPath from '../routing';
import * as events from '../api/sockets';

export function* playCard(socket, action) {
  yield call(events.playCard, socket, action.card)
}

export function* playerBegin(socket) {
  yield takeEvery('PLAY_CARD', playCard, socket);
  yield call(toPath.play);
}

export function* playerJoin(socket) {
  yield call(events.joinPlayersRoom, socket);
  yield takeEvery('BEGIN_GAME', playerBegin, socket);
  yield call(toPath.waiting);
}
