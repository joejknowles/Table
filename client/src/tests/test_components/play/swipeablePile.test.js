import { SwipeablePile } from '../../../components/play/swipeablePile';

import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
const shallowRenderer = ReactTestUtils.createRenderer();

it('SwipeablePile renders as before', () => {
  const component = shallowRenderer.render(<SwipeablePile cardCount={ 3 } />);
  expect(component).toMatchSnapshot();
});


it('SwipeablePile renders as before with one card', () => {
  const component = shallowRenderer.render(<SwipeablePile cardCount={ 1 } />);
  expect(component).toMatchSnapshot();
});
