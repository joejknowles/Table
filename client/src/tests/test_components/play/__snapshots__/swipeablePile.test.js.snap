exports[`test SwipeablePile renders as before 1`] = `
<SwipeCards
  className="pile">
  <Card
    onSwipeTop={undefined}>
    <div
      className="face-down" />
  </Card>
  <Card>
    <div
      className="face-down" />
  </Card>
</SwipeCards>
`;

exports[`test SwipeablePile renders as before with one card 1`] = `
<SwipeCards
  className="pile">
  <Card
    onSwipeTop={undefined}>
    <div
      className="face-down" />
  </Card>
</SwipeCards>
`;
